v2/call_option.go

```diff
@@ -173,6 +174,19 @@ func (o grpcOpt) Resolve(s *CallSettings) {
 	s.GRPC = o
 }
 
+type urlOpt struct {
+	u *url.URL
+}
+
+func (u urlOpt) Resolve(s *CallSettings) {
+	s.URL = u.u
+}
+
+// WithURL applies a URL override to the HTTP-based APICall.
+func WithURL(u *url.URL) CallOption {
```

Comment:
Our client option that lets people provide an endpoint override just lets them provide a string which is an easier type to work with. Maybe we should do the same here? Also, do we expect users to ever set this? I don't think so, and if not we should document as such. Maybe even make the name WithInternalURL to really try to tell people not to use it.

Reply:
I see your point, the `url.URL` might be too heavy-weight. So something like `WithPath(path string)` and expect something like "/v1beta/projects/my-project/foos/my-foo" i.e. a fully resolved path? Seems fine to me.

> do we expect users to ever set this?

Yeah I don't expect users to set this for GAPICs. I don't particularly like the "WithInternal" naming, so can we just document it as such? Maybe we should have an `internaloption` pkg like other modules have...

Reply:
(resolved offline) Just sticking with internal commentary.